base/tuple.jl

```diff
@@ -488,17 +488,14 @@ reverse(t::Tuple) = revargs(t...)
 
 ## specialized reduction ##
 
-# TODO: these definitions cannot yet be combined, since +(x...)
-# where x might be any tuple matches too many methods.
-# TODO: this is inconsistent with the regular sum in cases where the arguments
-# require size promotion to system size.
-sum(x::Tuple{Any, Vararg{Any}}) = +(x...)
+prod(x::Tuple{}) = 1
 
 # NOTE: should remove, but often used on array sizes
 # TODO: this is inconsistent with the regular prod in cases where the arguments
 # require size promotion to system size.
-prod(x::Tuple{}) = 1
-prod(x::Tuple{Any, Vararg{Any}}) = *(x...)
+_prod_simple(x) = prod(x)
```

Comment:
Is this method really needed? It seems to me in any case this is used, the caller could just call `prod` instead?

Reply:
These two functions are not equal, and the idea was to remove incorrect code from `prod` and clearly state that `_prod_simple` is not correct at edge cases for `Tuple`. 

First, I have tried it without this alias, but `_prod_simple` is sometimes called for an array as an argument in `abstractarray.jl`. Therefore `_prod_simple` just calls `prod` for any type except for `Tuple`. The new function seems necessary because `prod` is not yet included from `reduce.jl`, and bootstrap fails if removed.

Reply:
@JeffBezanson Thanks for the comment. I've found a way not to define a separate method and updated the PR accordingly. Now, it seems much cleaner, and both TODOs are fully resolved.